packages/frontend/src/components/common/PivotTable/index.tsx

```diff
@@ -34,6 +35,28 @@ const PivotTable: FC<PivotTableProps> = ({
     const { cx: tableCx, classes: tableStyles } = usePivotTableStyles();
     const { cx: cellCx, classes: cellStyles } = usePivotTableCellStyles({});
 
+    const getUnderlyingFieldValues = useCallback(
+        (rowIndex: number, colIndex: number) => {
+            const field = data.dataValues[rowIndex][colIndex];
+
+            const initialData =
+                field && field.value ? { [field.fieldId]: field.value } : {};
+
+            return [
+                // get the index values for this row
+                ...(data.indexValues[rowIndex] ?? []),
+                // get the header values for this column
+                ...(data.headerValues.map((hv) => hv[colIndex]) ?? []),
+            ]
+                .filter((iv) => iv.type === 'value')
+                .reduce<UnderlyingValueMap>((acc, iv) => {
+                    if (!iv.value) return acc;
+                    return { ...acc, [iv.fieldId]: iv.value };
+                }, initialData);
+        },
+        [data.indexValues, data.headerValues, data.dataValues],
+    );
```

Comment:
gets value types from both headers and indexes (dimensions with values)

I should probably move this to table context. I was also thinking maybe making a separate context for pivot tables?